data_log.py

```diff
@@ -486,6 +527,254 @@ def mean_absolute_percentage_error(y_true, y_pred):
+            # Move along one time step
+            start += 1
+            
+        # Convert list to np.array
+        X = np.asarray(X)
+        y = np.asarray(y)
+        
+        return X, y
+
+
+    data = dataframe.copy()
+    
+    if window_days*24 > data.values.shape[0]:
+        raise ValueError('Variable window_days has too large value: {}*24h = {} > {},             which is more than there is data!'.format(window_days, window_days*24, 
+                                                      data.values.shape[0]))
+    
+    # Training period
+    # ---------------
+    train_percent = train_percent/100.
+    st = pd.to_datetime(start_date)  # start date
+    et = st + dt.timedelta(days=int(train_percent*window_days))  # end date
+    train = data.loc[st:et].values
+    
+    # Standardize and transform training data set
+    mean_std_values = {}
+    for i, column in enumerate(columns):
+        # Calculate mean and standard deviation only
+        # from the training data set values
+        mu = train[:,i].mean()  # axis=0
+        sd = train[:,i].std()
+        mean_std_values[column] = (mu, sd)
+        # Standardize training data
+        train[:,i] = (train[:,i] - mu)/sd
+    
+    # Create overlapping windows with training data
+    X_train, y_train = overlap_windows(train, timesteps, steps_ahead)
+    
+    # Testing / Validation period
+    # ---------------------------
+    sv = et 
+    ev = sv + dt.timedelta(days=int((1-train_percent)*window_days)+1)
+    test = data.loc[sv:ev].values
+    
+    # Transform testing/validation data set
+    for i, column in enumerate(columns):
+        # Use mean and standard deviation from the
+        # training data set
+        mu = mean_std_values[column][0]
+        sd = mean_std_values[column][1]
+        # Standardize test data
+        test[:,i] = (test[:,i] - mu)/sd
+    
+    # Create overlaping windows with test data
+    X_test, y_test = overlap_windows(test, timesteps, steps_ahead)
+    
+    return mean_std_values, X_train, y_train, X_test, y_test
+
+
+# In[ ]:
+
+
+# Day-ahead (i.e. 24-hours ahead) short-term load prediction with time-series data
+START_DATE = '2014-01-09'  # starting date of the time-series data
+HISTORY_SIZE = 7*24        # window size in hours for the past history values
+STEPS_AHEAD = 24           # hours-ahead for prediction
+WINDOW_SIZE_DAYS = 300     # window size in days for training and validation
+
+
+# In[ ]:
+
+
+# Training, Validation and Testing data sets
+mean_std_values, X_train, y_train, X_test, y_test = samples_timesteps_features(
+    dataframe, ['Load', 'Temperature'], 
+    start_date=START_DATE, 
+    timesteps=HISTORY_SIZE, 
+    steps_ahead=STEPS_AHEAD,
+    window_days=WINDOW_SIZE_DAYS)
+
+
+# In[ ]:
+
+
+print('Training data shape:')
+print(X_train.shape, y_train.shape)
+print('Validation/Testing data shape:')
+print(X_test.shape, y_test.shape)
+
+
+# In[ ]:
+
+
+BATCH_SIZE = 256       # batch size
```

Comment:
Batch size and number of epochs can be optimized along with network architecture.